wdl/src/main/scala/wdl/AstTools.scala

```diff
@@ -304,21 +304,40 @@ object AstTools {
       * if terminal is "a" and the trail contains a MemberAccess(lhs: Terminal("a"), rhs: Terminal("b")),
       * this will return Some(Terminal("b"))
       */
-    lazy val terminalSubIdentifier: Option[Terminal] = trail.collectFirst {
+    private lazy val terminalSubIdentifier: Option[Terminal] = trail.collectFirst {
       case a: Ast if a.isMemberAccess
         && a.getAttribute("lhs") == terminal
         && a.getAttribute("rhs").isTerminal => a.getAttribute("rhs").asInstanceOf[Terminal]
     }
 
-    def fullVariableReferenceString: String = terminal.getSourceString + (terminalSubIdentifier map { "." + _.getSourceString } getOrElse "")
+    private lazy val fullVariableReferenceString: String = terminal.getSourceString + (terminalSubIdentifier map { "." + _.getSourceString } getOrElse "")
+
+    private def findResolvableSubstring(name: String, previous: String): Option[String] = {
+      lazy val popLastNamePiece: Option[String] = {
+        name.lastIndexOf(".") match {
+          case -1 => None
+          case i => Some(name.substring(0, i))
+        }
+      }
+
+      from.resolveVariable(name) match {
+        case Some(_: WdlTaskCall) => Some(previous)
+        case Some(_) => Some(name)
```

Comment:
Need to treat workflow calls like task calls too

Reply:
Yes and `Option(name)`.